File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        Dictionary<int, int> dict = new Dictionary<int, int>() ;
        for(int i =0 ; i< numbers.Length ; i++){
            int remain = target - numbers[i];
            if(dict.ContainsKey(remain)){
                return new int[]{dict[remain]+1, i +1};
            }else{
                dict[numbers[i]] = i ;
            }
        }
        return new int[]{-1,-1} ;
    }
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.java
class Solution {
    List<List<Integer>> res = new ArrayList<>();
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        Arrays.sort(candidates);
        process(candidates, 0, target, new ArrayList<Integer>());
        return res ;
    }

    void process(int[] candidates, int index, int remain, List<Integer> path){
        if(remain == 0){ 
            res.add( new ArrayList<>(path)) ;
            return ;
        }

        for(int i = index ; i < candidates.length ; i++){
            if(candidates[i] > remain){
                break;
            }

            if(i > index && candidates[i] == candidates[i-1]){
                continue;
            } 
            path.add(candidates[i]) ;
            process(candidates, i + 1, remain - candidates[i], path);
            path.remove(path.size() -1 );
            
        }
    }
}
